tests/functional/restricted.sh

```diff
@@ -13,9 +13,6 @@ nix-instantiate --restrict-eval ./simple.nix -I src1=simple.nix -I src2=config.n
 (! nix-instantiate --restrict-eval --eval -E 'builtins.readFile ./simple.nix')
 nix-instantiate --restrict-eval --eval -E 'builtins.readFile ./simple.nix' -I src=../..
 
-(! nix-instantiate --restrict-eval --eval -E 'builtins.readDir ../../src/nix-channel')
-nix-instantiate --restrict-eval --eval -E 'builtins.readDir ../../src/nix-channel' -I src=../../src
-
```

Comment:
Fishing out some non-test-suite files like this is no good with the broken-up build. If we want to test `..` we should do it a different way (e.g. make some dirs in `$TEST_ROOT` and `cd` inside there).